src/Aspire.Hosting/Dcp/Process/ProcessUtil.cs

```diff
@@ -49,67 +49,61 @@ public static (Task<ProcessResult>, IAsyncDisposable) Run(ProcessSpec processSpe
         {
             process.OutputDataReceived += (_, e) =>
             {
-                lock (processEventLock)
+                startupComplete.Wait();
```

Comment:
Comment why we need to wait

Reply:
The reason is all the same for all 3 handlers: I want the client to be able to finish handling OnStarted before any other events are reported. The idea being, in OnStarted there might be some code that sets up data structures that then are used for output processing and process exit handling. 

It is a guarantee that makes the client code much simpler to write.